base/linalg/triangular.jl

```diff
@@ -1900,14 +1907,10 @@ end
 function sqrtm{T}(A::UnitUpperTriangular{T})
```

Comment:
The `@inbounds` means that it never checks where `i` and `j` are in `1:n`.  It still checks whether `i ≤ j`, since if `i > j` then it just returns zero rather than looking at `A.data`.

Timing such a small operation is tricky; I would normally recommend BenchmarkTools instead of `@time`:
```julia
julia> using BenchmarkTools

julia> A = rand(10,10); T = UpperTriangular(A);

julia> @btime $A[3,5]; @btime $T[3,5];
  1.377 ns (0 allocations: 0 bytes)
  1.650 ns (0 allocations: 0 bytes)
```